language/tools/vm-genesis/src/lib.rs

```diff
@@ -510,29 +514,27 @@ fn reconfigure(session: &mut Session<StateViewCache>) {
 
 /// Verify the consistency of the genesis `WriteSet`
 fn verify_genesis_write_set(events: &[ContractEvent]) {
-    // Sanity checks on emitted events:
-    // (1) The genesis tx should emit 1 event: a NewEpochEvent.
+    // (1) first event is account creation event for LibraRoot
+    let create_libra_root_event = &events[0];
     assert_eq!(
-        events.len(),
-        //1, // This is the proper number of events for mainnet. Once we have a good layering
-        // strategy for mainnet/testnet genesis writesets uncomment this and remove the line
-        // below.
-        10, // XXX/TODO(tzakian). For testnet only!
-        "Genesis transaction should emit one event, but found {} events: {:?}",
-        events.len(),
-        events,
+        *create_libra_root_event.key(),
+        CreateAccountEvent::event_key(),
     );
 
-    // (2) The first event should be the new epoch event
-    let new_epoch_event = &events[0];
+    // (2) second event is account creation event for TreasuryCompliance
+    let create_treasury_compliance_event = &events[1];
     assert_eq!(
-        *new_epoch_event.key(),
-        new_epoch_event_key(),
-        "Key of emitted event {:?} does not match change event key {:?}",
-        *new_epoch_event.key(),
-        new_epoch_event_key(),
+        *create_treasury_compliance_event.key(),
+        CreateAccountEvent::event_key(),
     );
-    // (3) This should be the first new_epoch_event
+
+    // (3) The first non-account creation event should be the new epoch event
+    let new_epoch_event = events
+        .iter()
+        .find(|e| e.key() == &NewEpochEvent::event_key())
+        .expect("Couldn't find NewEpochEvent");
+    assert_eq!(*new_epoch_event.key(), NewEpochEvent::event_key());
```

Comment:
What would you think of adding an additional check that there's only one new epoch event?

Reply:
Very good idea, would not want to make that mistake (and it'd be easy to make!).